src/databricks/labs/ucx/progress/history.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+import dataclasses
+import datetime as dt
+import json
+import logging
+from collections.abc import Sequence
+from typing import ClassVar, Protocol, TypeVar, Generic, Any
+
+from databricks.labs.lsql.backends import SqlBackend
+from databricks.sdk import WorkspaceClient
+
+from databricks.labs.ucx.framework.owners import Ownership
+from databricks.labs.ucx.framework.utils import escape_sql_identifier
+from databricks.labs.ucx.progress.install import Historical
+
+logger = logging.getLogger(__name__)
+
+
+class DataclassInstance(Protocol):
+    __dataclass_fields__: ClassVar[dict[str, Any]]
+
+    __id_fields__: ClassVar[Sequence[str]]
```

Comment:
verify if this is not messing up with LSQL schema creation - check `deploy_schemas`